packages/two_dimensional_scrollables/lib/src/tree_view/tree_core.dart

```diff
@@ -0,0 +1,156 @@
+
+  /// Switches the given [TreeViewNode]s expanded state.
+  ///
+  /// May trigger an animation to reveal or hide the node's children based on
+  /// the [TreeView.animationStyle].
+  ///
+  /// If the node does not have any children, nothing will happen.
+  void toggleNode(TreeViewNode<T> node);
+
+  /// Closes all parent [TreeViewNode]s in the tree.
+  void collapseAll();
+
+  /// Expands all parent [TreeViewNode]s in the tree.
+  void expandAll();
+
+  /// Retrieves the [TreeViewNode] containing the associated content, if it
+  /// exists.
+  ///
+  /// If no node exists, this will return null. This does not reflect whether
+  /// or not a node [isActive], or if it is visible in the viewport.
+  TreeViewNode<T>? getNodeFor(T content);
+
+  /// Returns the current row index of the given [TreeViewNode].
+  ///
+  /// If the node is not currently active in the tree, meaning its parent is
+  /// collapsed, this will return null.
+  int? getActiveIndexFor(TreeViewNode<T> node);
+}
+
+/// Represents the animation of the children of a parent [TreeViewNode] that
+/// are animating into or out of view.
+///
+/// The [fromIndex] and [toIndex] are inclusive of the children following the
+/// parent, with the [value] representing the status of the current animation.
+///
+/// Provided to [RenderTreeViewport] by [TreeView] to properly offset animating
+/// children.
+typedef TreeViewNodesAnimation = ({
+  int fromIndex,
+  int toIndex,
+  double value,
+});
+
+/// Traversal order pattern for [TreeViewNode]s that are children of a
+/// [TreeView].
+enum TreeViewTraversalOrder {
+  /// Pre-order depth traversal.
+  ///
+  /// This traversal pattern will visit each given [TreeViewNode] before
+  /// visiting each of its children.
+  ///
+  /// This is the default traversal pattern for [TreeView.traversalOrder].
+  depthFirst,
+
+  /// Lever order traversal.
+  ///
+  /// This traversal pattern will visit each node that exists at the same
+  /// [TreeViewNode.depth], before progressing to the next depth of nodes in
+  /// the tree.
+  ///
+  /// Can be used in [TreeView.traversalOrder], which defaults to [depthFirst].
+  breadthFirst,
+}
+
+/// The style of indentation for [TreeViewNode]s in a [TreeView], as handled
+/// by [RenderTreeViewport].
+///
+/// By default, the indentation is handled by [RenderTreeViewport]. Child nodes
+/// are offset by the indentation specified by
+/// [TreeViewIndentationType.value] in the cross axis of the viewport. This
+/// means the space allotted to the indentation will not be part of the space
+/// made available to the Widget returned by [TreeView.treeNodeBuilder].
+///
+/// Alternatively, the indentation can be implemented in
+/// [TreeView.treeNodeBuilder], with the depth of the given tree row accessed
+/// by [TreeViewNode.depth]. This allows for more customization in building
+/// tree rows, such as filling the indented area with decorations or ink
+/// effects.
+class TreeViewIndentationType {
```

Comment:
Enhanced enums do not allow for user defined values, I did try that first.
I wanted to make this its own type, because I anticipate future feature requests around managing tree indentation.